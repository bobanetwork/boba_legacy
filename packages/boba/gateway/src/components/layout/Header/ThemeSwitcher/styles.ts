import styled, { css } from 'styled-components'
import { mobile } from 'themes/screens'

export const IconWrapper = styled.div`
  border-radius: 50%;
  cursor: pointer;
  &:hover {
    background: ${({ theme: { name, colors } }) =>
      name === 'light' ? colors.gray[400] : colors.gray[300]};
  }
  div {
    padding: 4px;
    height: 32px;
    width: 32px;
    display: flex;
    justify-content: center;
    align-items: center;
  }
  ${mobile(css`
    display: none;
  `)}
`
