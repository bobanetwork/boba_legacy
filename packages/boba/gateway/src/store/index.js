/*
Copyright 2021-present Boba Network.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

     http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License. */

import { createStore , applyMiddleware } from 'redux'
import RootReducer from 'reducers'
import reduxThunk from 'redux-thunk'
import persistReducer from 'redux-persist/lib/persistReducer'
import storage from 'redux-persist/lib/storage'
import persistStore from 'redux-persist/lib/persistStore'

const initialState = {}

const persistConfig = {
  key: 'root',
  storage,
  whitelist: ['network']
}

const persistedReducer = persistReducer(persistConfig, RootReducer)

export const store = createStore(
  persistedReducer,
  initialState,
  applyMiddleware(reduxThunk)
)

export default store;

export const persistor = persistStore(store)
