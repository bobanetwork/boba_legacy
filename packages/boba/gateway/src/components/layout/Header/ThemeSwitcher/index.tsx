import { Svg } from 'components/global'
import React, { FC } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { selectModalState } from 'selectors'

import { setTheme } from 'actions/uiAction'
import MoonIcon from 'assets/images/theme-moon.svg'
import SunIcon from 'assets/images/theme-sun.svg'
import { useTheme } from 'styled-components'
import { IconWrapper } from './styles'

enum THEME_NAME {
  LIGHT = 'light',
  DARK = 'dark',
}

const ThemeSwitcher: FC = () => {
  const currentTheme = useSelector(selectModalState('theme'))
  const dispatch = useDispatch<any>()

  const theme: any = useTheme()

  const onThemeChange = (themeName: string) => {
    localStorage.setItem('theme', themeName)
    dispatch(setTheme(themeName))
  }

  return (
    <>
      {currentTheme === THEME_NAME.LIGHT ? (
        <IconWrapper
          title="dark-icon"
          onClick={() => onThemeChange(THEME_NAME.DARK)}
        >
          <Svg src={MoonIcon} fill={theme.colors['gray'][600]} />
        </IconWrapper>
      ) : (
        <IconWrapper
          title="light-icon"
          onClick={() => onThemeChange(THEME_NAME.LIGHT)}
        >
          <Svg src={SunIcon} fill={theme.colors['gray'][100]} />
        </IconWrapper>
      )}
    </>
  )
}

export default ThemeSwitcher
