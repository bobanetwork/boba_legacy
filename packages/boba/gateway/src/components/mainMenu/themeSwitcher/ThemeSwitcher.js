import React from 'react'
import * as S from './ThemeSwitcher.styles.js'
import DarkIcon from 'components/icons/DarkIcon.js'
import LightIcon from 'components/icons/LightIcon.js'
import { setTheme } from 'actions/uiAction.js'
import { useSelector } from 'react-redux'
import { selectModalState } from 'selectors/uiSelector.js'
import { useDispatch } from 'react-redux'
import { IconButton } from '@mui/material'

function ThemeSwitcher() {
  const theme = useSelector(selectModalState('theme'));
  const dispatch = useDispatch();
  return (
    <S.ThemeSwitcherTag>
      {theme === 'light' ?
        <IconButton onClick={() => {
          localStorage.setItem('theme', 'dark');
          dispatch(setTheme('dark'));
        }} selected={true}>
          <DarkIcon />
        </IconButton>
        : <IconButton onClick={() => {
          localStorage.setItem('theme', 'light');
          dispatch(setTheme('light'));
        }} selected={true}>
          <LightIcon />
        </IconButton>}
    </S.ThemeSwitcherTag>
  );
}

export default ThemeSwitcher;
