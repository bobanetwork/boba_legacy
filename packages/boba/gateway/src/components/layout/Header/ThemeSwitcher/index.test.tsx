import { fireEvent, render, screen } from '@testing-library/react'
import React from 'react'
import CustomThemeProvider from 'themes'
import { Provider } from 'react-redux'
import { MemoryRouter } from 'react-router-dom'
import ThemeSwitcher from '.'
import store from 'store'

const renderThemeSwitcher = () => {
  return render(
    <MemoryRouter>
      <Provider store={store}>
        <CustomThemeProvider>
          <ThemeSwitcher />
        </CustomThemeProvider>
      </Provider>
    </MemoryRouter>
  )
}

describe('Layout => Header => ThemeSwitcher', () => {
  beforeEach(() => {
    jest.spyOn(Storage.prototype, 'setItem')
    Storage.prototype.setItem = jest.fn()
  })

  test('should match the snapshot', () => {
    const { asFragment } = renderThemeSwitcher()
    expect(asFragment()).toMatchSnapshot()
  })

  test('should switch themes & show icons correctly when clicked on button', () => {
    renderThemeSwitcher()

    expect(screen.queryByTitle('dark-icon')).toBeNull()
    expect(screen.getByTitle('light-icon')).toBeVisible()

    fireEvent.click(screen.getByTitle('light-icon'))
    expect(localStorage.setItem).toHaveBeenCalledWith('theme', 'light')

    expect(screen.getByTitle('dark-icon')).toBeVisible()
    expect(screen.queryByTitle('light-icon')).toBeNull()
  })
})
