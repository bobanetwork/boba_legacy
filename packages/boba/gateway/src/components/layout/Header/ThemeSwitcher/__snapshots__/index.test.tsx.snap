// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout => Header => ThemeSwitcher should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-fABZjn jWXFnH"
    title="light-icon"
  >
    <div>
      <div>
        <div
          data-src="theme-sun.svg"
          src=""
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
